### PR TITLE
Add project directory size to project show page via AJAX

### LINKED ISSUE #187: Display disk space usage per project
When worrying about disk space usage, it would be nice to see and/or sort jobs disk usage in human readable format. This would allow the user to see jobs that need to be archived or deleted to more quickly free up space.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1201737555416866) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by abujeda**
Possible implementation to display the project directory size in the project show page.
This is still a work in progress, but wanted to discuss if this PR points in the right direction.

It is implemented using AJAX to avoid performance penalties to the show page.
As well, a timeout is added to the backend call to mitigate problems with the backend request to get the directory size.

Fixes #187

### CODE REVIEW

**Comment by abujeda** [comment]
Demo:
project_size

**Comment by johrstrom** [comment]
Seems fine to me generally, though it's failing tests.  I only have a minor nit-pick.

File: apps/dashboard/app/models/project.rb [hunk 1]
```diff
@@ -90,13 +90,25 @@ def initialize(attributes = {})
...
     }
   end
 
+  def to_json(options = {})
+    project_size = size
+    json_data = to_h.merge(
+      {
+        :directory    => project_dataroot,
+        :sizeInBytes  => project_size,
+        :sizeInString => ::ApplicationController.helpers.number_to_human_size(project_size),
```
Comment (johrstrom) [inline apps/dashboard/app/models/project.rb:106]: I'm not in love with using camelCase. Can we use `size` and maybe `human_size`?

File: apps/dashboard/app/models/project.rb [hunk 2]
```diff
@@ -90,13 +90,25 @@ def initialize(attributes = {})
...
-      :name => name,
+      :id          => id,
+      :name        => name,
       :description => description,
-      :icon => icon,
+      :icon        => icon,
     }
   end
 
+  def to_json(options = {})
```
Comment (johrstrom) [inline apps/dashboard/app/models/project.rb:100]: OK - I guess I have something more meaty than a nitpick - We should probably be using the `jbuilder` utilities like we use elsewhere. 

Here you can see the view the files controller/models use when responding to json. 

https://github.com/OSC/ondemand/blob/master/apps/dashboard/app/views/files/index.json.jbuilder

Comment (abujeda) [reply apps/dashboard/app/models/project.rb]: Made some changes based on the feedback and fixed the failing tests.

File: apps/dashboard/app/views/projects/show.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,8 @@
+json.id @project.id
+json.name @project.name
+json.description @project.description
+json.icon @project.icon
+json.directory @project.directory
+project_size = @project.size
+json.size project_size
+json.human_size ::ApplicationController.helpers.number_to_human_size(project_size)
```
Comment (johrstrom) [inline apps/dashboard/app/views/projects/show.json.jbuilder:8]: since this is in a view now, can you just call `number_to_human_size` directly?

Comment (johrstrom) [reply apps/dashboard/app/views/projects/show.json.jbuilder:8]: Meh - even if you can, it's not that big of a deal.

**Review by johrstrom — APPROVED** [review]
LGTM

**Pull request merged**